hack/bundle-gen.py

```diff
@@ -0,0 +1,429 @@
+import urllib3
+import yaml
+
+# HIVE_VERSION_PREFIX is the prefix of the Hive version that will be constructed by this script
+# the version used for images and bundles will be "{prefix}.{number of commits}-{git hash}" eg. "v1.2.3187-18827f6"
+HIVE_VERSION_PREFIX = "v1.2"
+
+# Hive dir within both:
+# https://github.com/redhat-openshift-ecosystem/community-operators-prod
+# https://github.com/k8s-operatorhub/community-operators
+HIVE_SUB_DIR = 'operators/hive-operator'
+
+OPERATORHUB_HIVE_IMAGE_DEFAULT = 'quay.io/openshift-hive/hive'
+
+REGISTRY_AUTH_FILE_DEFAULT = '{}/.docker/config.json'.format(os.environ['HOME'])
+
+COMMUNITY_OPERATORS_HIVE_PKG_URL = 'https://raw.githubusercontent.com/redhat-openshift-ecosystem/community-operators-prod/main/operators/hive-operator/hive.package.yaml'
+
+CHANNEL_DEFAULT = 'alpha'
+BRANCH_DEFAULT = 'master'
+
+SUBPROCESS_REDIRECT = subprocess.DEVNULL
+
+def get_params():
+    parser = argparse.ArgumentParser(description='Hive Bundle Generator and Publishing Script')
+    parser.add_argument('--verbose',
+                        default=False,
+                        help='Show more details while running',
+                        action='store_true')
+    parser.add_argument('--dry-run',
+                        default=False,
+                        help='Test run that skips pushing branches and submitting PRs',
+                        action='store_true')
+    parser.add_argument('--branch',
+                        default=False,
+                        help='''The branch (or commit-ish) from which to build and push the image.
+                                If unspecified, we assume `master`. If we're using `master`, we
+                                generate the bundle version number based on ${HIVE_VERSION_PREFIX} and
+                                update the `alpha` channel. If BRANCH *also* corresponds to an `ocm-X.Y`,
+                                we'll also update that channel with the same bundle.If BRANCH corresponds
+                                to an `ocm-X.Y` but *not* `master`, we'll generate the bundle version
+                                number based on `X.Y` and update only that channel.''')
+    parser.add_argument('--registry-auth-file',
+                        default=REGISTRY_AUTH_FILE_DEFAULT,
+                        help='Path to registry auth file')
+    parser.add_argument('--github-user',
+                        default=os.environ['USER'],
+                        help="User's github username, if different than $USER")
+    parser.add_argument('--hold',
+                        default=False,
+                        help='Adds a /hold comment in commit body to prevent the PR from merging (use "/hold cancel" to remove)',
+                        action='store_true')
+    args = parser.parse_args()
+
+    if args.branch and args.branch == BRANCH_DEFAULT:
+        parser.error('--branch=master is assumed by this script and corresponds to the alpha channel')
```

Comment:
I don't know that we want to error if they specify the default explicitly. But if we're going to do this check, we need to be cleverer about detecting whether whatever commit-ish they specify corresponds to master.